image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
+When creating Docker images to run on OpenShift, there are a number of best practices to consider as an image author to ensure a good experience for consumers of those images. Because images are intended to be immutable and used as is, it is your responsibility to ensure:
 
-== General Docker Best Practices
-The following are some guidelines that apply to Docker image creation in general, independent of whether the images are used on OpenShift v3.
+- the image contains any artifacts that would typically be needed by a user of the image, and
+- the image can be configured by the user.
 
-=== Reuse images
-Whereever possible you should base your image on an appropriate upstream image via the `FROM` statement.  This will ensure that your image can easily pick up security fixes
-from an upstream image when it is updated, rather than you being responsible for updating your dependencies directly.  
+== General Docker Guidelines
+The following are some guidelines that apply to Docker image creation in general, independent of whether the images are used on OpenShift. See the official https://docs.docker.com/articles/dockerfile_best-practices/[Docker documentation] for more comprehensive guidelines.
 
-In addition, you should use tags in the `FROM` instruction (for example `rhel:rhel7`) will make it clear to users exactly which version of an image your image is based on.  If you use a tag other than `latest` this ensures that your image is not subjected to breaking changes that might go into the `latest` version of an upstream image.
+*Reusing Images*
 
-=== Tagging images
-When tagging your own images, you should strive to maintain backwards compatibility within a tag.  For example, if you provide an image named "foo" and it currently includes version 1.0, you might provide a tag of "foo:v1".  When you update the image, as long as it continues to be compatible with the orginal image, you can continue to tag the new image "foo:v1" and downstream consumers of this tag will be able to get updates without being broken.  If you come out with an incompatible update, then you should switch to a new tag, e.g. "foo:v2".  This allows downstream consumers to move up to the new version at will, but not be inadvertently broken by the new incompatible image.
+Wherever possible, base your image on an appropriate upstream image using the `FROM` statement. This ensures your image can easily pick up security fixes from an upstream image when it is updated, rather than you being responsible for updating your dependencies directly.
```

Comment:
[Suggestion for readabilty]
Wherever possible, we recommend that you base your image on an appropriate upstream image using the `FROM` statement. This ensures your image can easily pick up security fixes from an upstream image when it is updated, rather than you having to update your dependencies directly.